### PR TITLE
Feat/export

### PR DESCRIPTION
**Pull request by paulineribeyre**
https://ctds-planx.atlassian.net/browse/PXP-3602: improve required fields
https://ctds-planx.atlassian.net/browse/PXP-3598: add export button

### CODE REVIEW

File: src/api/users.js [hunk 1]
```diff
@@ -73,7 +87,7 @@ export const postUser = async (user, token) => {
         'Content-Type': 'application/json',
         'Authorization': `bearer ${accessToken}`
       },
-      body: JSON.stringify({username: user.username.trim(), name: user.name.trim(), eracommons: user.eracommons.trim(), orcid: user.orcid.trim(), organization: user.organization.trim(), contact_email: user.contact_email.trim(), google_email: user.google_email.trim(), expiration: user.expiration.trim(), datasets: user.datasets}),
+      body: JSON.stringify(getCleanUserData(user)),
```
Comment (rudyardrichter) [inline src/api/users.js]: :ok_hand:

File: src/components/UserInformation.js [hunk 2]
```diff
@@ -11,6 +11,11 @@ const usernameOptions = [
   { value: 'eracommons', label: 'eRA Commons ID' }
 ];
 
+// required for any user creation:
+const requiredFields = ['name', 'usernameOption'];
+// required when a DAC creates a PI:
+const requiredDacFields = ['organization', 'expiration'];
```
Comment (rudyardrichter) [inline src/components/UserInformation.js]: it seems like these two are only used together, is there a reason to have them separated?

File: src/components/UserInformation.js [hunk 3]
```diff
@@ -144,7 +154,7 @@ class UserInformation extends React.Component {
         <ul className='form-info__details'>
           <h2>User Details</h2>
           <li className='form-info__detail'>
-            <label>ACCESS Username</label>
+            <label>ACCESS Username {this.isFieldRequired('usernameOption') ? '*' : ''}</label>
```
Comment (rudyardrichter) [inline src/components/UserInformation.js]: could add a separate function to annotate so you don't have to repeat this, maybe like
```js
const annotateRequired = (field, text) => this.isFieldRequired(field) ? text + "*" : text
// ...
<label>{annotateRequired('usernameOption', 'ACCESS Username')</label>
```
what do you think?

File: src/components/UserInformation.js (hunk 2 above)
Comment (paulineribeyre) [reply src/components/UserInformation.js]: ah, right, this doesn't need to be separate anymore

**Pull request merged**